python/kvikio/kvikio/_lib/arr.pyx

```diff
@@ -203,7 +207,7 @@ cdef class Array:
         for i in range(self.ndim):
             o = self.shape_mv[i]
             Py_INCREF(o)
-            PyTuple_SET_ITEM(shape, i, o)
+            PyTuple_SetItem(shape, i, o)
```

Comment:
Is the incref still needed here?

Same question occurs with similar lines below

Reply:
It looks like both `PyTuple_SET_ITEM` and `PyTuple_SetItem` [both steal the reference](https://docs.python.org/3/c-api/tuple.html). That means that 1) nothing has changed relative to the old behavior in this PR, and 2) this incref is indeede necessary since we are retaining local references to `o` here and without the incref we will wind up undercounting the number of references alive.

Reply:
Ok great. Thank you for checking! 🙏

Remember there being subtle behavior around stealing with these functions

Agree this seems reasonable then